ddl/table.go

```diff
@@ -265,6 +265,14 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		return ver, errors.Trace(err)
 	}
 
+	// Update the table's auto increment ID.
+	if newSchemaID != oldSchemaID {
+		_, err = t.GenAutoTableID(newSchemaID, tblInfo.ID, baseID)
```

Comment:
When running this DDL, user insert a bulk of data into the table. So some tidb-sever may allocate ID larger than current baseID.

Reply:
@shenli If others update the autoTableID base, this ddl txn will fail cause by key changed?

Reply:
I think line 255 should change to `err = t.DropTable(oldSchemaID, tblInfo.ID, true)`, It will be more reasonable. But the issue you mention will not happen, because allocate ID is got from GenAutoTableID, which will hinc the old autoID key, and the dll job will failed cause by the key changed.

Reply:
Okay, I got it. if this ddl commit firstly, maybe in other tidb will get the old autoID key because the schema is not syncd in this tidb.